### ISSUE TITLE
Specify directories for deps, devDeps

### ISSUE THREAD

**Issue by kumavis** [depcheck/depcheck#530]
getting false negatives (depcheck passes but theres actually a problem) because deps are listed as devDeps but actually needed for prod

if i could specify dir/files to parse as deps and dir/files to parse as devDeps it would be solve this problem

```bash
npx depcheck --deps ./src --devDeps ./test
```

**Comment by rumpl** [comment]
Hey @kumavis 

We actually had something like this at one time and decided to remove it see #120 #121 

Can't remember exactly when and how this was a problem for us.

What you could do is [run depcheck in a script](https://github.com/depcheck/depcheck#api) and look at `unused.using`. I'll need some time to think about the best way (or if we want to) to handle your usecase

**Comment by kumavis** [comment]
@rumpl interesting, is there more context somewhere?
https://github.com/depcheck/depcheck/issues/114 doesnt really explain what was broken

**Comment by rumpl** [comment]
@lijunle do you remember what problems this had?

If I had to redo it, it would be a `--production` flag, same as npm WDYT?

**Comment by lijunle** [comment]
I think the problem is, we cannot skip the calculation of devDep because we need that information to compare with `package.json` to generate the missing list. If we want to *ingore* devDep, we can only do a display hide at the last step.

**Comment by kumavis** [comment]
yeah i guess if you're specifying which files to parse for deps and devDeps, the resultant "missing" lists are more numerous:
- missing
- unused
- shouldBeDev
- shouldBeProd

**Comment by lijunle** [comment]
@kumavis Let me step back, why do you want to check a subset of files? If you are doing so, do you still care about "missing" list?

IMO, compare the information from a subset of files with project level `package.json` metadata is not making sense.

**Comment by kumavis** [comment]
@lijunle 
the issue I had was:
- a required prod dep was set as a devDep
- depcheck passed
- failed in prod as the dep was missing

**Comment by lijunle** [comment]
It sounds like the problem is, depcheck does not have a way to know which files are for prod and which for testing.

AFAIK, ESLint use the [directory hierarchy RC files](https://eslint.org/docs/user-guide/configuring#configuration-cascading-and-hierarchy) to solve this kind of problem. I am not sure if that model fits here.

**Comment by kumavis** [comment]
> It sounds like the problem is, depcheck does not have a way to know which files are for prod and which for testing.

yes, and perhaps a significant feature to implement, but does not seem out of scope for `depcheck`

**Comment by TimBailey-pnk** [comment]
Hi, I have a simialr issue.
I have my test files as siblings of the real files, ie
```
./lib/datastore.js
./lib/datastore.spec.js
```
What I would like is that if the dependency is ONLY used in *.spec.js then its a `devDependency` and should warn if it is not

What do you think?
